src/MoBi.Core/Domain/Repository/BuildingBlockRepository.cs

```diff
@@ -27,10 +28,16 @@ public IEnumerable<IBuildingBlock> All()
 
       public IEnumerable<TBuildingBlock> All<TBuildingBlock>() where TBuildingBlock : IBuildingBlock
       {
-         if (_context.CurrentProject == null)
+         var currentProject = _projectRetriever.Current;
+         if (currentProject == null)
             return new List<TBuildingBlock>();
 
-         return _context.CurrentProject.AllBuildingBlocks().OfType<TBuildingBlock>();
+         return currentProject.AllBuildingBlocks().OfType<TBuildingBlock>().Concat(moduleBuildingBlocks<TBuildingBlock>(currentProject));
```

Comment:
It's going to be some work to remove the underlying _buildingBlocks list in MoBi.

I'll use #972 to rework this, and all the interaction tasks that use it.

For now, this should patch it for the purposes of working with simulation validations.